### PR TITLE
Site Transfers: restyle Domains section on transfer screen

### PR DESCRIPTION
**Pull request by katinthehatsite**
Closes https://github.com/Automattic/dotcom-forge/issues/2625

## Proposed Changes

This PR restyles **Domains** section on the transfer site screen to be more consistent with Figma design file: 1U910ArebC7gbLF1RtxUxE-fi-1_2.

**Before**:

Screenshot 2023-06-19 at 4 04 45 PM

**After**: 

Screenshot 2023-06-19 at 4 03 43 PM

## Testing Instructions

* Make sure you have a site with at least one custom domain
* Navigate to **Settings > General > Transfer your site**: https://wpcalypso.wordpress.com/settings/start-site-transfer/{yoursite}
* Select an administrator to transfer your site to
* Proceed to the transfer screen
* Observe the design of the "Domains" section

**Note**: In my PR, the domain icons are not fully aligned with the text but this was the best I could do with them. If you have any suggestions on aligning them better, let me know. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx [hunk 1]
```diff
@@ -62,6 +62,23 @@ const ListItem = styled.li( {
...
+	paddingBottom: '1em',
+} );
+
+const DomainsWrapperItem = styled.div( {
+	paddingLeft: '1em',
+	fontWeight: 500,
+} );
+
+const GridiconStyled = styled( Gridicon )( {
+	paddingRight: '10px',
```
Comment (wojtekn) [inline client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx:79]: `vertical-align: middle;` should do the trick.

Comment (sejas) [reply client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx:79]: This works for me.
Another option would be setting : `display: "flex"` to the parent: `DomainsWrapperItem`.

**Review by sejas — APPROVED** [review]
Working as expected. Just a vertical alignment issue with the domain icon.

File: client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx (hunk 1 above)
Comment (katinthehatsite) [reply client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx:79]: thanks for the pointers!

**Review by wojtekn — APPROVED** [review]

**Pull request merged**